magento/magento2
# magento/magento2#38659: Change order status issue

- Fixed issue with changing order status





### Description (*)


### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#38659
2. Fixes magento/magento2#38917

### Manual testing scenarios (*)

1. Install vanilla Magento
3. Go to admin Stores => Order Status
4. Create two statuses "Sent" and "Ready for Pickup" and bind them to "Processing" state
5. Create product and register customer
6. Place order on front using this product and customer and default shipping and payment methods
7. Go to admin Sales => Orders
8. Open previously placed order
9. Create invoice by clicking Invoice button on top
10. See order changed status to "Processing"
11. Go to "Add Comment" form change status to "Sent" and add any comment
12. Reload order form and see status changed to "Sent"
13. Go to "Add Comment" form again and change status to "Ready for Pickup" and add any comment
14. Reload order form

### Questions or comments


### Contribution checklist (*)
